application/service/factory/service_factory.go

```diff
@@ -167,6 +168,16 @@ func WithClusterService(s service.ClusterService) Option {
 	}
 }
 
+// WithAuthenticationProviderService overrides the default function that returns the AuthenticationProviderService,
+// so that instead, a mock implementation can be used
+func WithAuthenticationProviderService(s service.AuthenticationProviderService) Option {
```

Comment:
This is not the preferred way to mock this, now that we have factories.  We can leave it for now though and address it later.

Reply:
yes, I've just used the way we mock `WITService`, but we should definitely align everything once this PR is merged